### PR TITLE
fix(snap): Refactor to avoid conflicts with readonly config provider directory

### PR DESCRIPTION
**Pull request by MonicaisHer**
Changes to incorporate the following fixes:
- https://github.com/canonical/edgex-snap-hooks/pull/83
- https://github.com/canonical/edgex-snap-hooks/pull/82

For testing instructions, please refer to https://github.com/edgexfoundry/device-mqtt-go/pull/535

### CODE REVIEW

File: snap/local/helper-go/install.go [hunk 1]
```diff
@@ -39,7 +40,14 @@ func installConfig() error {
 func install() {
 	log.SetComponentName("install")
 
-	if err := installConfig(); err != nil {
-		log.Fatalf("Error installing config files: %s", err)
+	// Install default config files only if no config provider is connected
+	isConnected, err := snapctl.IsConnected(app + "-config").Run()
```
Comment (farshidtz) [inline snap/local/helper-go/install.go:44]: The identifier of the config interface slot is not kuiperd-config.

File: snap/local/helper-go/main.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ import (
 	"os"
 )
 
+const app = "kuiperd"
```
Comment (farshidtz) [inline snap/local/helper-go/main.go:22]: Having a global variable called `app` is confusing because this snap has two apps.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/local/helper-go/main.go (hunk 2 above)
Comment (MonicaisHer) [reply snap/local/helper-go/main.go]: It has been updated to `const daemonApp = "kuiperd"`.

**Comment by MonicaisHer** [comment]
- [ ] Rerun the snap testing once this [PR](https://github.com/canonical/edgex-snap-testing/pull/171) has been merged.

File: snap/local/bin/source-env-file.sh [hunk 3]
```diff
@@ -1,13 +1,13 @@
-#!/bin/bash -e
+#!/bin/bash -ex
```
Comment (farshidtz) [inline snap/local/bin/source-env-file.sh:1]:
```suggestion
#!/bin/bash -e
```

I think if xtrace is enabled, the commands will appear in syslogs without the appropriate component name or tag. The reader can't know which application they come from.

File: snap/local/bin/source-env-file.sh [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 
 SERVICE="kuiperd"
-SERVICE_ENV="$SNAP_DATA/config/$SERVICE/res/$SERVICE.env"
+ENV_FILE="$SNAP_DATA/config/$SERVICE/res/overrides.env"
```
Comment (farshidtz) [inline snap/local/bin/source-env-file.sh:4]:
```suggestion
ENV_FILE="$SNAP_DATA/config/$SERVICE/overrides.env"
```

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/local/bin/source-env-file.sh (hunk 4 above)
Comment (farshidtz) [reply snap/local/bin/source-env-file.sh:4]: Please note that there are no tests for the config options passed to this snap. So the tests pass regardless.

**Review by farshidtz — APPROVED** [review]
Looks good. Thanks

**Pull request merged**